### PR TITLE
Move musl internal lock functions from JS to C

### PR DESCRIPTION
**Pull request by sbc100**
__lockfile/__unlockfile were already patched to support `__EMSCRIPTEN__`
in system/lib/libc/musl/src/stdio/__lockfile.c so just start compiling
that file.

__lock/__unlock were implemented as no-ops in two different places
so move them to extras.c for now.

The fact that both these pairs of functions are currently nop'd out
is almost cetainly a problem for pthreads builds, but not one that
is solved by this PR :)

### CODE REVIEW

File: tools/system_libs.py [hunk 1]
```diff
@@ -832,13 +833,13 @@ def can_build(self):
 
 class libc_extras(MuslInternalLibrary):
```
Comment (kripken) [inline tools/system_libs.py]: maybe let's call it `extras_fastcomp` like the filename?

Comment (sbc100) [reply tools/system_libs.py]: This was always a fastcomp-only library.   I'm not changing that about it.   We could rename it but I'm not sure it worth it at this point with fastcomp going away.

**Comment by sbc100** [comment]
It looks like __lock and __unlock are used quite bit so we probably do want real implementation of them at some point.   For example here are the users of __unlock:

```
system/lib/libc/musl/src/dirent/readdir_r.c:            UNLOCK(dir->lock);
system/lib/libc/musl/src/dirent/readdir_r.c:    UNLOCK(dir->lock);
system/lib/libc/musl/src/dirent/rewinddir.c:    UNLOCK(dir->lock);
system/lib/libc/musl/src/dirent/seekdir.c:      UNLOCK(dir->lock);
system/lib/libc/musl/src/exit/at_quick_exit.c:          UNLOCK(lock);
system/lib/libc/musl/src/exit/at_quick_exit.c:  UNLOCK(lock);
system/lib/libc/musl/src/exit/atexit.c:         UNLOCK(lock);
system/lib/libc/musl/src/exit/atexit.c:                 UNLOCK(lock);
system/lib/libc/musl/src/exit/atexit.c: UNLOCK(lock);
system/lib/libc/musl/src/internal/libc.h:#define UNLOCK(x) __unlock(x)
system/lib/libc/musl/src/internal/stdio_impl.h:#define FUNLOCK(f) if (__need_unlock) __unlockfile((f)); else
system/lib/libc/musl/src/locale/dcngettext.c:                   UNLOCK(lock);
system/lib/libc/musl/src/locale/dcngettext.c:   UNLOCK(lock);
system/lib/libc/musl/src/locale/locale_map.c:                   UNLOCK(lock);
system/lib/libc/musl/src/locale/locale_map.c:   UNLOCK(lock);
system/lib/libc/musl/src/locale/setlocale.c:            UNLOCK(lock);
system/lib/libc/musl/src/locale/setlocale.c:    UNLOCK(lock);
system/lib/libc/musl/src/misc/syslog.c: UNLOCK(lock);
system/lib/libc/musl/src/misc/syslog.c: UNLOCK(lock);
system/lib/libc/musl/src/misc/syslog.c: UNLOCK(lock);
system/lib/libc/musl/src/misc/syslog.c: UNLOCK(lock);
system/lib/libc/musl/src/prng/random.c: UNLOCK(lock);
system/lib/libc/musl/src/prng/random.c: UNLOCK(lock);
system/lib/libc/musl/src/prng/random.c: UNLOCK(lock);
system/lib/libc/musl/src/prng/random.c: UNLOCK(lock);
system/lib/libc/musl/src/stdio/clearerr.c:      FUNLOCK(f);
system/lib/libc/musl/src/stdio/fclose.c:        else FUNLOCK(f);
system/lib/libc/musl/src/stdio/feof.c:  FUNLOCK(f);
system/lib/libc/musl/src/stdio/ferror.c:        FUNLOCK(f);
system/lib/libc/musl/src/stdio/fflush.c:                FUNLOCK(f);
system/lib/libc/musl/src/stdio/fflush.c:                FUNLOCK(f);
system/lib/libc/musl/src/stdio/fgetln.c:        FUNLOCK(f);
system/lib/libc/musl/src/stdio/fgets.c:         FUNLOCK(f);
system/lib/libc/musl/src/stdio/fgets.c: FUNLOCK(f);
system/lib/libc/musl/src/stdio/fgetwc.c:        FUNLOCK(f);
system/lib/libc/musl/src/stdio/fgetws.c:        FUNLOCK(f);
system/lib/libc/musl/src/locale/locale_map.c:                   UNLOCK(lock);
system/lib/libc/musl/src/locale/locale_map.c:   UNLOCK(lock);
system/lib/libc/musl/src/locale/setlocale.c:            UNLOCK(lock);
system/lib/libc/musl/src/locale/setlocale.c:    UNLOCK(lock);
system/lib/libc/musl/src/misc/syslog.c: UNLOCK(lock);
system/lib/libc/musl/src/misc/syslog.c: UNLOCK(lock);
system/lib/libc/musl/src/misc/syslog.c: UNLOCK(lock);
system/lib/libc/musl/src/misc/syslog.c: UNLOCK(lock);
system/lib/libc/musl/src/prng/random.c: UNLOCK(lock);
system/lib/libc/musl/src/prng/random.c: UNLOCK(lock);
system/lib/libc/musl/src/prng/random.c: UNLOCK(lock);
system/lib/libc/musl/src/prng/random.c: UNLOCK(lock);
system/lib/libc/musl/src/stdio/clearerr.c:      FUNLOCK(f);
system/lib/libc/musl/src/stdio/fclose.c:        else FUNLOCK(f);
system/lib/libc/musl/src/stdio/feof.c:  FUNLOCK(f);
system/lib/libc/musl/src/stdio/ferror.c:        FUNLOCK(f);
system/lib/libc/musl/src/stdio/fflush.c:                FUNLOCK(f);
system/lib/libc/musl/src/stdio/fflush.c:                FUNLOCK(f);
system/lib/libc/musl/src/stdio/fgetln.c:        FUNLOCK(f);
system/lib/libc/musl/src/stdio/fgets.c:         FUNLOCK(f);
system/lib/libc/musl/src/stdio/fgets.c: FUNLOCK(f);
system/lib/libc/musl/src/stdio/fgetwc.c:        FUNLOCK(f);
system/lib/libc/musl/src/stdio/fgetws.c:        FUNLOCK(f);
system/lib/libc/musl/src/stdio/fileno.c:        FUNLOCK(f);
system/lib/libc/musl/src/stdio/fputwc.c:        FUNLOCK(f);
system/lib/libc/musl/src/stdio/fputws.c:                        FUNLOCK(f);
system/lib/libc/musl/src/stdio/fputws.c:        FUNLOCK(f);
system/lib/libc/musl/src/stdio/fread.c:                 FUNLOCK(f);
system/lib/libc/musl/src/stdio/fread.c: FUNLOCK(f);
system/lib/libc/musl/src/stdio/freopen.c:       FUNLOCK(f);
system/lib/libc/musl/src/stdio/fseek.c: FUNLOCK(f);
system/lib/libc/musl/src/stdio/ftell.c: FUNLOCK(f);
system/lib/libc/musl/src/stdio/fwide.c: FUNLOCK(f);
system/lib/libc/musl/src/stdio/fwrite.c:        FUNLOCK(f);
system/lib/libc/musl/src/stdio/getdelim.c:              FUNLOCK(f);
system/lib/libc/musl/src/stdio/getdelim.c:                              FUNLOCK(f);
system/lib/libc/musl/src/stdio/getdelim.c:      FUNLOCK(f);
system/lib/libc/musl/src/stdio/getdelim.c:      FUNLOCK(f);
system/lib/libc/musl/src/stdio/ofl.c:   UNLOCK(ofl_lock);
system/lib/libc/musl/src/stdio/perror.c:        FUNLOCK(f);
system/lib/libc/musl/src/stdio/popen.c:                         FUNLOCK(f);
system/lib/libc/musl/src/stdio/puts.c:  FUNLOCK(stdout);
system/lib/libc/musl/src/stdio/rewind.c:        FUNLOCK(f);
system/lib/libc/musl/src/stdio/ungetc.c:                FUNLOCK(f);
system/lib/libc/musl/src/stdio/ungetc.c:        FUNLOCK(f);
system/lib/libc/musl/src/stdio/ungetwc.c:               FUNLOCK(f);
system/lib/libc/musl/src/stdio/ungetwc.c:       FUNLOCK(f);
system/lib/libc/musl/src/stdio/vfprintf.c:      FUNLOCK(f);
system/lib/libc/musl/src/stdio/vfscanf.c:       FUNLOCK(f);
system/lib/libc/musl/src/stdio/vfwprintf.c:     FUNLOCK(f);
system/lib/libc/musl/src/stdio/vfwscanf.c:      FUNLOCK(f);
system/lib/libc/musl/src/thread/pthread_atfork.c:               UNLOCK(lock);
system/lib/libc/musl/src/thread/pthread_atfork.c:       UNLOCK(lock);
system/lib/libc/musl/src/thread/sem_open.c:             UNLOCK(lock);
system/lib/libc/musl/src/thread/sem_open.c:             UNLOCK(lock);
system/lib/libc/musl/src/thread/sem_open.c:     UNLOCK(lock);
system/lib/libc/musl/src/thread/sem_open.c:     UNLOCK(lock);
system/lib/libc/musl/src/thread/sem_open.c:     UNLOCK(lock);
system/lib/libc/musl/src/thread/sem_open.c:     UNLOCK(lock);
system/lib/libc/musl/src/thread/synccall.c:     UNLOCK(synccall_lock);
system/lib/libc/musl/src/time/__tz.c:                   UNLOCK(lock);
system/lib/libc/musl/src/time/__tz.c:   UNLOCK(lock);
system/lib/libc/musl/src/time/__tz.c:   UNLOCK(lock);
system/lib/libc/musl/src/time/__tz.c:   UNLOCK(lock);
system/lib/libc/musl/src/time/__tz.c:   UNLOCK(lock);

```

**Comment by sbc100** [comment]
Even FFLOCK and FUNLOCK seem to protect the library-side data structure, not just so much serialize the syscalls.   Assuming thats what you mean?

For example `fgets` uses to lock to control access to data in FILE* structure.

File: tools/system_libs.py (hunk 1 above)
Comment (kripken) [reply tools/system_libs.py]: Makes sense, yeah, no point in polishing fastcomp stuff.

**Comment by kripken** [comment]
Oh, interesting, I didn't realize musl internally might parallelize its own I/O data structures. Then yeah, we should fix that in pthreads builds. I suspect this isn't a problem in practice since we've never encountered an error, but now I'm not sure... I opened #10595 so we don't forget.

**Pull request merged**